sway/server.c

```diff
@@ -278,6 +279,9 @@ bool server_init(struct sway_server *server) {
 	wlr_scene_set_gamma_control_manager_v1(root->root_scene,
 		server->gamma_control_manager_v1);
 
+	server->session_active.notify = handle_session_active;
+	wl_signal_add(&server->session->events.active, &server->session_active);
+
```

Comment:
I think we could run into a event ordering problem. DRM will watch the same event which will trigger a scanning of connectors. What happens if this runs first?

Reply:
It means that if connector state had changed since we were active last, we'd end up having to modeset multiple times in a row. I considered adding a new signal for ordering, but it felt annoying..

In the end, the new solution without needing to listen for session signals is much nicer.